cdc/puller/sorter/heap_sorter.go

```diff
@@ -146,6 +155,21 @@ func (h *heapSorter) flush(ctx context.Context, maxResolvedTs uint64) error {
 	if !isEmptyFlush {
 		backEndFinal := backEnd
 		err := heapSorterIOPool.Go(ctx, func() {
+			failpoint.Inject("asyncFlushStartDelay", func() {
+				log.Debug("asyncFlushStartDelay")
+			})
+
+			h.canceller.EnterAsyncOp()
+			defer h.canceller.FinishAsyncOp()
```

Comment:
If read lock is acquired, the `lock` operation will wait, does it mean the flush can't be cancelled during it is running, is it by design?

Reply:
Will fix